relay/sources/relayd/src/shared_files.rs

```diff
@@ -119,23 +122,33 @@ pub fn parse_value(key: &str, file: &str) -> Result<String, ()> {
 }
 
 pub fn same_hash_than_in_nodeslist(
-    hash_value: &str,
-    source_uuid: String,
-    job_config: Arc<JobConfig>,
-) -> bool {
-    let myvec2: Vec<String> = job_config
-        .nodes
-        .read()
-        .expect("Cannot read nodes list")
-        .get_keyhash_from_uuid(&source_uuid)
-        .unwrap()
-        .split(':')
-        .map(|s| s.to_string())
-        .collect();
+    pubkey: String,
+    hash_type: String,
+    keyhash: String,
+) -> Result<bool, Error> {
+    let pubkey_pem = get_pubkey(pubkey);
+
+    let rsa_key_from_pem = Rsa::public_key_from_pem_pkcs1(pubkey_pem.as_bytes()).unwrap();
+
+    let public_key_from_pem = PKey::from_rsa(rsa_key_from_pem)?;
```

Comment:
no need for this step, you should be able to `public_key_to_der` directly on the Rsa